### ISSUE TITLE
Rename class team to class side

### ISSUE THREAD

**Issue by hexagonrecursion** [wesnoth/wesnoth#5501]
### Describe the desired feature
The class named "team" should have really been named "side".
https://github.com/wesnoth/wesnoth/blob/c97d2cb6fb9ff4f8c6ad221b5c7b3922f40a954d/src/team.hpp#L40-L44
This situation is confusing. I want to gradually go through the source and replace "team" with "side" where appropriate. Is this a good idea?

**Comment by CelticMinstrel** [comment]
I'm not sure. It _is_ a bit confusing, but on the other hand, I'm not sure I like calling it "side" either… it feels a bit too vague for a class name.

**Comment by Pentarctagon** [comment]
game_side maybe?

**Comment by hexagonrecursion** [comment]
Coming from the wml land the terms "side" and "team" had established and very specific meanings for me. It was *super* confusing reading the code until I realized that in the C++ land the terms are not used consistently.

**Comment by CelticMinstrel** [comment]
Yeah, it's definitely confusing that a "side" in WML is a "team" in C++, while a "team" in WML is… I don't even know what it is in C++.

**Comment by soliton-** [comment]
`scenario_side` may be more accurate?

Would definitely be nice to clean this up.

**Comment by hexagonrecursion** [comment]
It is simply "side" in wml. It is true that out of context the word "side" has too many meanings, but I would argue that in this case it is in context (mostly due to "side" having a clear and established meaning in wml). That being said if you prefer "scenario_side" I'm fine with that option too.

**Comment by soliton-** [comment]
It might be better to be more specific in C++. Though I don't really mind `side` either since I agree as you say it's quite clearly established in WML.

**Comment by Pentarctagon** [comment]
I'm fine with `scenario_side` as well, for what it's worth.

**Comment by AI0867** [comment]
WML land:
side = a group of units sharing a treasury and flag
team = a group of sides that aren't hostile to each other

C++ land:
side = WML side numbers, 1-indexed
team = the object holding everything about a side
team_index = side number - 1, used as an index into the array of teams: 0-indexed

**Comment by Wedge009** [comment]
Establishing a consistent set of terminology is probably a good idea. However, I seem to recall numerous reported issues in the past where the 'side/team' flag in the game UI (top bar), will not be properly translated because of technical difficulties (possibly pop-up text upon mouse hover). Can't recall if it ever got resolved properly. Just keep in mind this bit when revising the text.

**Comment by Wedge009** [comment]
The question was brought up again (and probably has been many times before this) in [#wesnoth-dev](https://irclogs.wesnoth.org/2022/02/%23wesnoth-dev.2022-02-16.log):
```
20220216 09:23:39<+wDiscord> [RufDerMacht] I just got to know "SideWML". Are there any ambitions to rename this unfortunate, misleading designation to something like "PartyWML" or so? Or am I misunderstanding something again?
20220216 09:25:27<+wDiscord> [RufDerMacht] I mean, of course, the entire term "side".
20220216 09:28:02<+wDiscord> [octalot] Somewhere there's a suggestion to take all of the C++ classes and functions called `team` or `side` and name them consistently, but I don't see the WML `[side]` tag ever getting renamed.
20220216 09:28:25<+wDiscord> [octalot] Renaming it would break every scenario of every campaign.
20220216 09:32:30<+wDiscord> [max_torch] teams contain sides. I actually fixed this term that was inconsistently used in the ingame tutorial text a while back. It was a usage of 'team' which was actually referring to 'side'. [edited]
20220216 09:42:34<+wDiscord> [RufDerMacht] I see. I just thought at the WML level, not at C++ level. A solution could be, to establish a new term, that should be used from now then, and let the old term stay for compatibility. But after all, it's just a matter of habit. thx. [edited]
20220216 09:48:02< Soliton> what is the confusion in WML? what do you confuse side with?
```

**Comment by AI0867** [comment]
I have a half-finished and not-compiling branch locally, where I've attempted to bring C++-land naming in line with WML-land naming. I don't think I'll have the time to finish it anytime soon, but I can push what I have somewhere where it can be worked on further if anyone wants to.

**Comment by AI0867** [comment]
As it's been years, I'll drop the (still partial) patch here in case anyone wants to continue working on it.

[team-side.patch](https://github.com/user-attachments/files/26986765/team-side.patch)
